File: scratch/parse-text-stdin.go
package main

import "fmt"
import "bufio"
import "os"
import "regexp"

type liner struct {
    ip       []string
    time     string
    hostname []string
}

func mapLine(s string, filters map[string]*regexp.Regexp) *liner, bool {
    var profile liner
    var matched bool
    for name, regex := range filters {
        match := regex.FindAllString(s, -1)
        fmt.Printf("Regex '%s' matched to '%s'\n", name, match)
        switch name {
        case "ipaddr":
            profile.ip = match
            matched = true
        case "hostname":
            profile.hostname = match
            matched = true
        case default:
            matched = false
        } 
    }
    return (&profile, matched)
}


func main() {
    filters := make(map[string]*regexp.Regexp)
    filters["ipaddr"] = regexp.MustCompile(`\b(?:[0-9]{1,3}\.){3}[0-9]{1,3}\b`)
    filters["hostname"] = regexp.MustCompile(`hostname: [a-z0-9\-]*`)

    var linecount int
    var matchcount int

    input := bufio.NewScanner(os.Stdin)
    for input.Scan() {
        line := input.Text()
        linecount++
        // fmt.Println(reIP.MatchString(line))
        profile, matched := *mapLine(line, filters)
        if matched {
            matchcount++
        }
        fmt.Printf("IP: %s, hostname: %s\n\n", profile.ip, profile.hostname)
        // fmt.Printf("\t%s\n", line)
        if err := input.Err(); err != nil {
            fmt.Fprintf(os.Stderr, "parse-text-stdin: %v\n", err)
        }
    }
    fmt.Println("Lines: %d", linecount)
    fmt.Println("matches: %d", matchcount)
}

File: scratch/parse-text-stdin.py
import sys
import re
import time


def read_input(file):
    for line in file:
        yield line

def check_matches(line, regexes):
    match = False
    for name,r in regexes.items():
        m = re.search(r,line)
        if m:
            match = True
            '''
            if name == 'ipaddr':
                print("matched ipaddr: %s" % m.group())
            elif name == 'hostname':
                print("matched hostanme: %s" % m.group())
            '''
    return(match)


def main():
    start = time.time()
    re1 = re.compile('\b(?:[0-9]{1,3}\.){3}[0-9]{1,3}\b')
    re2 = re.compile('hostname: [a-z0-9\-]*')
    matchcount = 0
    linecount = 0
    regexes = {'ipaddr':re1, 'hostname': re2}
    data = read_input(sys.stdin)
    for thing in data:
        linecount += 1
        matched = check_matches(thing, regexes)
        if matched:
            matchcount += 1
    print("lines: %d" % linecount)
    print("matches: %d" % matchcount)
    end = time.time()
    print("timetaken: %f ms" % ((end-start)*1000))


if __name__ == "__main__":
    main()
